src/Scripting/Core/Hosting/ObjectFormatter/CommonTypeNameFormatter.cs

```diff
@@ -93,7 +93,7 @@ public virtual string FormatTypeArguments(Type[] typeArguments, CommonTypeNameFo
 
             if (typeArguments.Length == 0)
             {
-                return "";
+                throw new ArgumentException(nameof(typeArguments));
```

Comment:
Since this is effectively an internal type, might be just fine to replace these exceptions with Debug.Assert.